sns_ik_lib/include/sns_ik/sns_ik_math_utils.hpp

```diff
@@ -70,14 +70,14 @@ bool pinv_P(const Eigen::MatrixXd &A, Eigen::MatrixXd *invA, Eigen::MatrixXd *P,
  * @param A: input matrix
  *   FIXME: if A.rows() >= A.cols() causes a failed assertion, but this should be valid input
  * @param[out] invA: pseudoinverse of A
- * @param[in/out] P: the nullspace projector matrix:  P = (P - pinv(A)*A)
+ * @param[in/out/opt] P: the nullspace projector matrix:  P = (P - pinv(A)*A)
  *                   P.rows() == P.cols() = A.cols() is required
  *                   P should be initialized with the identity matrix
  * @param[opt] lambda_max: damping parameter for the damped pseudoinverse
  * @param[opt] eps: singular values smaller than this will be set to zero
  * @return: true if A is full rank, false if A is rank deficient
  */
-bool pinv_damped_P(const Eigen::MatrixXd &A, Eigen::MatrixXd *invA, Eigen::MatrixXd *P,
+bool pinv_damped_P(const Eigen::MatrixXd &A, Eigen::MatrixXd *invA, Eigen::MatrixXd *P = nullptr,
```

Comment:
Note for the future: we should be consistent about how we use references and pointers in function signatures. Also, when this is going to 18.04, let's move towards using `std::unique_ptr`'s instead of raw pointers.